source/text_handler.cpp

```diff
@@ -217,36 +213,63 @@ spv_result_t AssemblyContext::binaryEncodeU32(const uint32_t value,
   return SPV_SUCCESS;
 }
 
-spv_result_t AssemblyContext::binaryEncodeU64(const uint64_t value,
-                                              spv_instruction_t* pInst) {
-  uint32_t low = uint32_t(0x00000000ffffffff & value);
-  uint32_t high = uint32_t((0xffffffff00000000 & value) >> 32);
-  binaryEncodeU32(low, pInst);
-  binaryEncodeU32(high, pInst);
-  return SPV_SUCCESS;
-}
-
 spv_result_t AssemblyContext::binaryEncodeNumericLiteral(
     const char* val, spv_result_t error_code, const IdType& type,
     spv_instruction_t* pInst) {
   using spvutils::EncodeNumberStatus;
-  EncodeNumberStatus rc = EncodeNumberStatus::kSuccess;
+  // Populate the NumberType from the IdType for parsing.
+  spvutils::NumberType number_type;
+  switch (type.type_class) {
+    case IdTypeClass::kOtherType:
+      return diagnostic(SPV_ERROR_INTERNAL)
+             << "Unexpected numeric literal type";
+    case IdTypeClass::kScalarIntegerType:
+      if (type.isSigned) {
+        number_type = {type.bitwidth, spv_number_kind_t::SPV_NUMBER_SIGNED_INT};
```

Comment:
Using the type name to qualify the SPV_NUMBER_SIGNED_INT looks fishy to me.  This isn't a C++ enum class.  I'm not sure all compilers will accept this.
Please remove the type name qualifier.